src/util/transition/ngbCollapseTransition.ts

```diff
@@ -0,0 +1,63 @@
+import {NgbTransitionStartFn} from './ngbTransition';
+import {reflow} from '../util';
+
+export interface NgbCollapseCtx {
+  direction: 'show' | 'hide';
+  maxHeight?: string;
+}
+
+function measureCollapsingElementHeightPx(element: HTMLElement): string {
+  const {classList} = element;
+  const hasShownClass = classList.contains('show');
+  if (!hasShownClass) {
+    classList.add('show');
+  }
+
+  element.style.height = '';
+  const height = element.getBoundingClientRect().height + 'px';
+
+  if (!hasShownClass) {
+    classList.remove('show');
+  }
+
```

Comment:
Just a detail here, but the 'show' string (for the classname) can be factorized in a constant.

Reply:
it would not make the code more readable at all.